flink-core/src/main/java/org/apache/flink/api/connector/sink2/InitContext.java

```diff
@@ -57,6 +83,34 @@ public interface InitContext {
     /**
      * The ID of the current job. Note that Job ID can change in particular upon manual restart. The
      * returned ID should NOT be used for any job management tasks.
+     *
+     * @deprecated This method is deprecated since Flink 1.19. All metadata about the job should be
+     *     provided uniformly by {@link #getJobInfo()}.
+     * @see <a
+     *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+     *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
      */
     JobID getJobId();
+
+    /**
+     * Get the meta information of current job.
+     *
+     * @return the job meta information.
+     */
+    @PublicEvolving
+    @Nullable
+    default JobInfo getJobInfo() {
+        return null;
+    }
+
+    /**
+     * Get the meta information of current task.
+     *
+     * @return the task meta information.
+     */
+    @PublicEvolving
+    @Nullable
+    default TaskInfo getTaskInfo() {
+        return null;
+    }
```

Comment:
Same here for default and nullable.

Reply:
I've changed the default methods including `getJobInfo()` and `getTaskInfo()` to be abstract.